exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/entities/CompositeIdTableEntityTest.kt

```diff
@@ -0,0 +1,556 @@
+    }
+
+    class Review(id: EntityID<CompositeID>) : CompositeEntity(id) {
+        companion object : CompositeEntityClass<Review>(Reviews)
+
+        var book by Book referencedOn Reviews.book
+    }
+
+    private val allTables = arrayOf(Publishers, Authors, Books, Reviews)
+
+    // The tests below will be removed before merging ------------------------------------------------------------
+    // They only exist in case the branch is checked out, to view composite PK types, etc.
+
+    @Test
+    fun entityIdUseCases() {
+        withTables(excludeSettings = listOf(TestDB.SQLITE), tables = allTables) {
+            // entities
+            val publisherA: Publisher = Publisher.new {
+                name = "Publisher A"
+            }
+            val authorA: Author = Author.new {
+                publisher = publisherA
+                penName = "Author A"
+            }
+            val bookA: Book = Book.new {
+                title = "Book A"
+                author = authorA
+            }
+            val reviewIdValueOg = CompositeID {
+                it[Reviews.content] = "Not bad"
+                it[Reviews.rank] = 12345
+            }
+            val reviewA: Review = Review.new(reviewIdValueOg) {
+                book = bookA
+            }
+
+            // entity id properties
+            val publisherId: EntityID<CompositeID> = publisherA.id
+            val authorId: EntityID<Int> = authorA.id
+            val bookId: EntityID<CompositeID> = bookA.id
+            val reviewId: EntityID<CompositeID> = reviewA.id
+
+            // access wrapped entity id values
+            val publisherIdValue: CompositeID = publisherId.value
+            val authorIdValue: Int = authorId.value
+            val bookIdValue: CompositeID = bookId.value
+            val reviewIdValue: CompositeID = reviewId.value
+            assertEquals(reviewIdValueOg, reviewId.value)
+
+            // access individual composite entity id values - these are now each wrapped as EntityID
+            val publisherIdComponent1: EntityID<Int> = publisherIdValue[Publishers.pubId]
+            val publisherIdComponent2: EntityID<UUID> = publisherIdValue[Publishers.isbn]
+            val bookIdComponent1: EntityID<Int> = bookIdValue[Books.bookId]
+            val reviewIdComponent1: EntityID<String> = reviewIdValue[Reviews.content]
+            val reviewIdComponent2: EntityID<Long> = reviewIdValue[Reviews.rank]
+
+            // access individual composite values - requires unwrapping - no type erasure
+            val publisherIdComponent1Value: Int = publisherIdValue[Publishers.pubId].value
+            val publisherIdComponent2Value: UUID = publisherIdValue[Publishers.isbn].value
+            val bookIdComponent1Value: Int = bookIdValue[Books.bookId].value
+            val reviewIdComponent1Value: String = reviewIdValue[Reviews.content].value
+            val reviewIdComponent2Value: Long = reviewIdValue[Reviews.rank].value
```

Comment:
Something to note about using `entityId()` is that, because the existing behavior hasn't been changed, each composite PK column will be of type `Column<EntityID<T>>`. So while an `IntIdTable` only has one column with this type, its `id` column, a `CompositeIdTable` will have the `id` column (representative of all component references), as well as X amount more for X components.

Like with `IntIdTable`, I've made it so that manually defining a new ID doesn't require wrapping with `EntityID`, but accessing the original value will still require chaining `.value`.

So with an `IntIdTable` we could access the individual value like -> `id.value`
With a `CompositeIdTable` this means doing -> `id.value[Table.column].value` for each individual value